### ISSUE TITLE
Ublock origins Extention not updating

### ISSUE THREAD

**Issue by reaper666735** [MrAlex94/Waterfox-Old#28]
**Describe the bug**

Hello, as the tittle is self explanatory, Ublock seems to be "stuck" in an older version (1.32.4) that was released in January of this year i never noticed till now .... , when i try to look for updates 'there are none' or go to the Github page and manually "force" an update to the latest 1.39.0 from 1 week ago it still remains the same after restart :(

no other extention seems to be having this issue (mobile included) so i wonder if this is something wrong with my WF ?

**Screenshots**

https://imgbox.com/Y2W7xtxj

https://imgbox.com/B3gdUfpw

https://imgbox.com/5DE2qJZN

**Desktop (please complete the following information):**
 - OS: Win 7 64bit
 - Version 2021.10 64bit

**Comment by Killogy** [comment]
Yes, It has been happening for quite some time, there are other extensions with which the same thing happens, for example https://addons.mozilla.org/es/firefox/addon/open-with/

**Comment by hawkeye116477** [comment]
Well, latest uBO webextension requires Firefox 60 and Waterfox Classic is based on FF 56. As I know, minimum version was bumped, cuz it requires now support for JS Modules and it's from FF 60. However Waterfox Classic already has support for that, so if you will install it manually from AMO, then probably everything will be working fine, but you won't get any updates and support.

So if you don't want to install updates for uBO by manual, then just use [uBO Legacy](https://github.com/gorhill/uBlock-for-firefox-legacy/releases). However unfortunately non-webext is less updated, hasn't many features known from webext and mainly relies on external contributors (see Issues tab).

**Comment by reaper666735** [comment]
excuse me but what is AMO  ?  ^^ 

and is there no way for WF Classic to have a certain "marker" to allow it to update normally as with FF 60+ ?, since as you stated it already has the JS module support required so just passing the installation 'check' is the hurdle i take it ?

**Comment by hawkeye116477** [comment]
> what is AMO?

It's `addons.mozilla.org`.

> and is there no way for WF Classic to have a certain "marker" to allow it to update normally as with FF 60+

I don't know if you can change that somehow by myself. Maybe that could be changed somewhere in code by @MrAlex94, but not every addon will work which is marked as compatible with FF 60, cuz some addons may use webextensions API introduced after FF 57, which Waterfox Classic doesn't support, some APIs are ported, but not every. So changing that could possibly cause problems with another extensions.

**Comment by walmar** [comment]
Hi @reaper666735 and others,
I use the following workaround since this happened.

1. Subscribe to news feed: [Release notes from uBlock](https://github.com/gorhill/uBlock/releases.atom)
2. Update to latest version:
2.1 about:addons > Extensions > uBlock Origin > Disable
2.2 [uBlock Origin – Get this Extension for 🦊 Firefox](https://addons.mozilla.org/firefox/addon/ublock-origin/)
   * v1.33.0–1.52.0 Download file > Add
   * v1.53.0+ Right click *Download file* to choose *Save Link As...* from the context menu to get the XPI installer module to your disk.
      1. Open the downloaded zip file *ublock_origin-1.xx.x.xpi* with a file archiver of your choice like [7-Zip](https://www.7-zip.org/).
      2. Drag the file *manifest.json* to extract it to a location of your choice like your desktop. Don't close the window of the file archiver.
      3. Open the extracted file with an editor of your choice like Notepad and remove the following content from it:
        ```js
        ,
            "gecko_android": {
              "strict_min_version": "79.0"
            }
        ```
      4. Save the changes, drag the edited file to the opened window of the file archiver and accept the query to update the archive.
      5. Drag the modified file *ublock_origin-1.xx.x.xpi* to the Waterfox window for installation process.
      6. Then you can delete the above named files.
   * v1.59.0 seems to require a modified *js/logger-ui.js* file with replacements from v1.57.0 as included in the attached file [ublock_origin-1.59.0WxMod.zip](https://github.com/user-attachments/files/17799599/ublock_origin-1.59.0WxMod.zip) to avoid *SyntaxError: bad method definition in logger-ui.js:236:11*
   * ❌v1.60.0+ does not display the user interface

   2.3 about:addons > Extensions > uBlock Origin > Enable

**Comment by reaper666735** [comment]
@walmar

this actually worked ty !, and i'm sure this will mean i will have to redo this operation every time a new 'major' release comes around :p 

anyway if MrAlex94 has any good idea's to help the code around that would be good ^^

**Comment by Felipefpl** [comment]
A better way to do this is following the releases here https://github.com/gorhill/uBlock/releases ;)

**Comment by walmar** [comment]
I suggest contributors to implement the following concept to close this issue.

- [ ] Show the user a suitable update notification.
- Like: New available [Extension] v[x.x.x] update for Firefox [xx]+ is potentially incompatible with Waterfox Classic.
- Similar to new updates from extensions with automatic update option to off.

**Comment by walmar** [comment]
I updated my [issuecomment-982883945](https://github.com/WaterfoxCo/Waterfox-Classic/issues/28#issuecomment-982883945).

**Comment by walmar** [comment]
I updated my [issuecomment-982883945](https://github.com/WaterfoxCo/Waterfox-Classic/issues/28#issuecomment-982883945).
